### PR TITLE
Feature/update xcuitest upload

### PR DESCRIPTION
**Pull request by sapieneptus**
### WIP

# Motivation

Support this format:
```Shell
$ xtc xcuitest  <api_key> --devices a1801a3c --user jenkins_qa@nonexistingemail.com --workspace ~/Somewhere/Build/Products/Debug-iphoneos/
```

And this:

```Shell
$ xtc xcuitest  <api_key> --app-file myApp.ipa --devices a1801a3c --user jenkins_qa@nonexistingemail.com --workspace ~/Somewhere/Build/Products/Debug-iphoneos/
```

# Note

To use this format, the command is `xcuitest`.

# TODO:

- [ ] Refactor unit tests, need to create new `UploadXCUITestCommandOptions.cs` file
- [ ] Test that xcuitests are not supported using `test` command
- [ ] Test that `--app-file` takes precedence over whatever AUT is in the workspace
- [ ] Some sanity tests on the `ArchiveBundle` function to check that directories exist etc.

### CODE REVIEW

**Comment by sapieneptus** [comment]
CC: @jonstoneman

**Comment by sapieneptus** [comment]
I've made some updates, see description.

**Comment by sapieneptus** [comment]
This is now in `WIP` status because I want to write more unit tests but the logic is generally there so review is welcome.

**Comment by sapieneptus** [comment]
@john7doe ready for re-review

File: Xtc/src/Xtc.TestCloud/Commands/UploadXCUITestsCommandOptions.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        public override void Validate()
+        {
+            ValidateRequiredOptions();    
+        }
+
+        private string _appFile;
+
+        public override string AppFile
+        {
+            get
```
Comment (john7doe) [inline Xtc/src/Xtc.TestCloud/Commands/UploadXCUITestsCommandOptions.cs]: not sure we really need to cache the file in _appFile. It make the code more complex, with very little gain. 

If we _must_ I would perfer to use Lazy<T>: https://msdn.microsoft.com/en-us/library/dd997286(v=vs.110).aspx and then cache the value regardless of it coming from AppFileOptionFlag or the workspace.

File: Xtc/src/Xtc.TestCloud/ObjectModel/XCUITestWorkspace.cs [hunk 2]
```diff
@@ -16,6 +17,15 @@ public class XCUITestWorkspace : IWorkspace
     {
         private readonly string _workspacePath;
 
+        public static string FindAUT(string workspacePath)
+        {
+            var regex = new Regex("^*[^-Runner].app$");
```
Comment (john7doe) [inline Xtc/src/Xtc.TestCloud/ObjectModel/XCUITestWorkspace.cs]: perfer Regex.IsMatch(String, String) or

Comment (john7doe) [inline Xtc/src/Xtc.TestCloud/ObjectModel/XCUITestWorkspace.cs]: looks good! comments not blocking in any way

File: Xtc/src/Xtc.TestCloud/Commands/UploadXCUITestsCommandOptions.cs (hunk 1 above)
Comment (sapieneptus) [inline Xtc/src/Xtc.TestCloud/Commands/UploadXCUITestsCommandOptions.cs]: What would you suggest as an alternate approach to caching the file?

The obvious alternative to me is that `UploadXCUITestsCommandOptions` doesn't inherit from `UploadTestsCommandOptions` and redeclares all the arguments etc., which IMO is a larger complication.

Comment (john7doe) [inline Xtc/src/Xtc.TestCloud/Commands/UploadXCUITestsCommandOptions.cs]: Looking at it again, its fine as it is.

**Pull request merged**